dot/state/block.go

```diff
@@ -250,14 +250,62 @@ func (bs *BlockState) GetHashByNumber(num uint) (common.Hash, error) {
 	return common.NewHash(bh), nil
 }
 
+// GetAllDescendants gets all the descendants even if hash is not stored in memory.
+func (bs *BlockState) GetAllDescendants(hash common.Hash) ([]common.Hash, error) {
```

Comment:
We need a unit test for this new function

Reply:
added the test.